### ISSUE TITLE
LayerNormalization with rms_scaling documentation is different from implementation

### ISSUE THREAD

**Issue by mzhukova** [keras-team/keras#21234]
The documentation mentions that 
> rms_scaling: If True, center and scale are ignored, and the inputs are scaled by gamma and the inverse square root of the square of all inputs. This is an approximate and faster approach that avoids ever computing the mean of the input.

However, in the [implementation](https://github.com/keras-team/keras/blob/master/keras/src/layers/normalization/layer_normalization.py#L207), it actually does the following:
```
        if self.rms_scaling:
            # Calculate outputs with only variance and gamma if rms scaling
            # is enabled
            # Calculate the variance along self.axis (layer activations).
            variance = ops.var(inputs, axis=self.axis, keepdims=True)
            inv = ops.rsqrt(variance + self.epsilon)

            outputs = (
                inputs * inv * ops.cast(_broadcast(self.gamma), inputs.dtype)
            )
```
So the mean is indeed used, as variance is computed here rather than RMS norm.

There was also a discussion during the addition of RMS Normalization (https://github.com/keras-team/keras/pull/20911#issuecomment-2687658774) that confirms this behavior.

I think the docs could use an update to clarify this behavior. Right now, it sounds like the mean isn't used when `rms_scaling` is on, but the code suggests otherwise.

**Comment by mzhukova** [comment]
Alternatively, the implementation could be adjusted to match the docs, though that might make RMSNormalization behave just like LayerNormalization with `rms_scaling=True`, correct?

**Comment by sonali-kumari1** [comment]
Hi @mzhukova - 

Thanks for pointing this out. The [documenation](https://keras.io/api/layers/normalization_layers/layer_normalization/) suggests that when` rms_scaling=true`, the layer performs normalization without involving mean of the inputs, similar to `RMS` normalization in this [paper](https://arxiv.org/pdf/1910.07467) which removes mean calculation and relies on `RMS` to reduce the computational overhead introduced by `LayerNorm`. However, as confirmed by #20911, it uses 
```
variance = var(inputs, axis=self.axis, keepdims=True)
inv = rsqrt(variance + epsilon)
outputs = inputs * inv * gamma
```
which means it still involves mean(since variance depends on the mean) and it is not true `RMS` normalization. We will look into this and update you.

**Comment by abheesht17** [comment]
Thanks for spotting this! Using `ops.var()` is definitely not the intended behaviour for `rms_scaling` and seems like a bug; when `rms_scaling = True`, the layer should function like RMSNorm (https://github.com/keras-team/keras-core/issues/715).

Also, the actual `RMSNormalization` layer will be a duplicate of `LayerNormalization` once we do the fix.

Will open a PR shortly to fix this.

Edit: Just raising a warning if `rms_scaling = True`. The correct way of using RMS norm is to use the `RMSNormalization` layer.

**Comment by rashmivemuri** [comment]
Hi Keras team, I tried reproducing issue (#21234) to update the docstring for LayerNormalization.
Could not reproduce the erroneous documentation (dated 1st May, 2025) or the code given in the issue containing variance-based normalisation. Probably because the implementation has been updated with mean-free RMS Normalisation. However, i also noticed that the new docstring is not aligned with the new implementation as it now points back to the variance-based approach, which is obsolete.

**Comment by hertschuh** [comment]
This is now working as designed after https://github.com/keras-team/keras/pull/21352

`RMSNormalization` is the supported way. `rms_scaling=True` is deprecated and hidden. We did NOT change the behavior of `rms_scaling=True` for backwards compatibility in case people rely on the numerics.
